ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java

```diff
@@ -175,4 +201,11 @@ private static SchemaResult notCompatible(
             + "Schema:" + schema,
         cause));
   }
+
+  private static String getSubject(final String topicName, final boolean isKey) {
```

Comment:
nit: I used this pattern quite a bit, maybe we should move it to a util and refactor this? (and then we can also make `SCHEMA_REGISTRY_*_SUFFIX` private and just use this method) can be done in follow-up PR

Reply:
I only see two uses of `SCHEMA_REGISTRY_KEY_SUFFIX` in the codebase so I've extracted those into a util method and made the constant private. However, there are a bunch of other uses of `SCHEMA_REGISTRY_VALUE_SUFFIX` that I want to leave for now to make it easier to audit to ensure proper handling of key schemas (at least parity with value schemas) so I've left those in for now. We can finish this refactor in a follow-up PR once the other bits of key schema support are in.